### PR TITLE
Use buy and sell methods of Celo exchange contract

### LINKED ISSUE #792: Make wallet exchange more predictable
### Expected Behavior

~We have seen a few instances of exchange transactions reverted in the wallet. We should update params to make this more reliable.~

EDIT: see conversation below.

### PR DESCRIPTION
**Pull request by Pedro-vk**
### Description

_Use buy and sell methods of Cello contracts doing the exchange more accurate_

### Tested

_Tested on Android 11. Going to Celo section, performing buy and sells, and checking the amount on Blockscout to ensure that the Celo is the amount typed on the exchange input._

### How others should test

_Buy and sell Celo and check it on Blockscout._

### Related issues

- Fixes #792

### CODE REVIEW

**Comment by etuleu** [comment]
Looks like some tests are failing. Also would be good to figure out what metrics we want to look at that would be a good indicator of the exchange working.

File: packages/mobile/src/exchange/saga.ts [hunk 1]
```diff
@@ -243,7 +258,7 @@ export function* exchangeGoldAndStableTokens(action: ExchangeTokensAction) {
     const exchangeContract: ExchangeWrapper = yield call(getExchangeContract, stableToken)
 
     const convertedMakerAmount: BigNumber = roundDown(
-      yield call(convertToContractDecimals, makerAmount, makerToken),
+      yield call(convertToContractDecimals, inputAmount, makerToken),
```
Comment (gnardini) [inline packages/mobile/src/exchange/saga.ts:261]: Why this change? The `makerAmount` is the amount of the token we'll sell, and the `inputAmount` is in the input which could be CELO or local currency, right?

File: packages/mobile/src/exchange/saga.ts [hunk 2]
```diff
@@ -300,15 +320,15 @@ export function* exchangeGoldAndStableTokens(action: ExchangeTokensAction) {
...
+    if (!isStableToGold) {
       approveTx = goldTokenContract.approve(
         exchangeContract.address,
         convertedMakerAmount.toString()
       )
     } else {
       approveTx = stableTokenContract.approve(
         exchangeContract.address,
-        convertedMakerAmount.toString()
+        convertedTakerAmount.multipliedBy(10 ** 20).toString()
```
Comment (gnardini) [inline packages/mobile/src/exchange/saga.ts:331]: Why are you multiplying by this number here?

**Review by gnardini — COMMENTED** [review]
Looks great, just left a couple questions :)

File: packages/mobile/src/exchange/saga.ts (hunk 1 above)
Comment (Pedro-vk) [reply packages/mobile/src/exchange/saga.ts:261]: Yes, but I used `inputAmount` because the maker amount comes in the incorrect token, I mean:
- Buying Celo the `makerAmount` is:
  - Using Celo as input: you buy 1 and you get 1.
  - Using stable as input: you buy 1 and you get the conversion of 1. So it is less accurate.

File: packages/mobile/src/exchange/saga.ts (hunk 2 above)
Comment (Pedro-vk) [reply packages/mobile/src/exchange/saga.ts:331]: ops! it was just a test, I'm going to clean it

**Comment by jeanregisser** [comment]
Closing, we'll revisit later. See https://github.com/valora-inc/wallet/issues/792#issuecomment-984682806
